src/components/register/RegisterComponent.js

```diff
@@ -19,121 +124,129 @@ const RegisterComponent = () =>
-                            <option value="WA">Washington</option>
-                            <option value="WI">Wisconsin</option>
-                            <option value="WV">West Virginia</option>
-                            <option value="WY">Wyoming</option>
+                        <label for="inputRegion" class="col-form-label text-white float-left">Region</label>
+                        <select id="inputRegion" class="form-control" 
+                            onChange={(e) => {
+                                let selectedRegion = e.target.value;
+                                let subregionList = this.getSubregionsForRegion(this.state.selectedContinent, selectedRegion);
+                                let selectedSubregion = subregionList[0][1];
+                    
+                                this.setState(prevState => (
+                                    {
+                                        ...prevState,
+                                        'selectedRegion': selectedRegion,
+                                        'subregionList': subregionList,
+                                        'selectedSubregion': selectedSubregion
+                                    }))}}>
+                            {this.state.regionList && this.state.regionList.map(regionKey => 
+                                <option key={regionKey} value={regionKey}>{regionKey}</option>
+                            )}
                         </select>
                     </div>
 
-                    <div class="form-group col-md-2">
-                        <label for="inputZip" class="col-form-label text-white float-left">Zip</label>
-                        <input type="text" class="form-control" id="inputZip" placeholder="Zip"/>
+                    <div class="form-group col-md-4">
+                        <label for="inputSubregion" class="col-form-label text-white float-left">Subregion</label>
+                        <select id="inputSubregion" class="form-control" 
+                            onChange={(e) => {
+                                this.updateField("selectedSubregion", e.target.value)
+                            }}
+                        >
+                            {this.state.subregionList && this.state.subregionList.map(subregionKey => 
+                                <option key={subregionKey[0]} value={subregionKey[1]}>{subregionKey[0]}</option>
+                            )}
+                        </select>                    
                     </div>
 
                 </div>
-
                 
 
                 <br/>
 
                 <div className="form-group row">
-                    <button className="btn btn-block btn-success">Register</button>
+                    <button className="btn btn-block btn-success"
+                        onClick={(e) => {
+                            e.preventDefault();//FIXME: Temporary workaround Prevents refreshing of the page (in order to show alert.)
```

Comment:
Ah you might've explained this to me in our last call, but do you have a plan set for a better way to get the error messages to show up? either way, any way i can help?